### ISSUE TITLE
implicit security check in Jupyter sim_apis

### ISSUE THREAD

**Issue by rorour** [radiasoft/sirepo#7026]
[Both of the APIs in sirepo/sim_api/jupyterhublogin.py begin with a call ](https://github.com/radiasoft/sirepo/blob/57ea8a13deafb49a42f9d3217275fd1126d0737d/sirepo/sim_api/jupyterhublogin.py#L32
) to `self.parse_params()`. These are the only calls in Sirepo to `self.parse_params()` that do not utilize the return value; they are intended to check that the user has the sim_type_jupyterhublogin role. These APIs depend on `Quest.parse_params` calling `http_request.parse_params` which calls `auth.check_sim_type_role` inside of a nested function. There is no documentation on either `parse_params` function indicating that they do a security check. 

At the very least, I want to add a comment above these calls in the Jupyter APIs indicating that these are security checks. Or, just change them to direct calls to `auth.check_sim_type_role`.

But I don't like that this check needs to be added inside each API. It seems like this check should be done at a higher level, like as an APIPerm `require_user_with_role`. What are your thoughts? @robnagler

**Comment by robnagler** [comment]
Yes, this is a real problem and #5026 hints at the general issue. Ideally, we'd always know the sim_type in every request, if there was one, and then could check it in check_api_call. However, we don't have that. It possibly could be added to the header of HTTP requests, and explicitly in the SIM requests. An alternative is to put this in template.import_module, but that's probably too late.

What we need to do is push on quest.Spec so that we can know to call parse_params, parse_post, etc. before the function is called.
